### PR TITLE
DAOS-4310 srv: bump ULTs stack-size to 32K

### PR DESCRIPTION
**Pull request by bfaccini**
To prevent cases where old PMDK RPMs (previous to
DAOS-4183 fix) we build, or PMDK local build done
without "NDCTL_DISABLE=y", are used. This is because
ndctl-based PMDK version will inconditionnally try
to check for bad blocks and thus call udev_new()
routine which allocates a 16K buffer on stack leading
to a stack overflow finaly causing corruption in
adjacent ULT stack.
This is the only known case of ULT stack overflow in
DAOS presently, and this patch is only an empirical
way to fix it by bumping ULT stack size, but a more
generic mechanism is required to at least detect any
cases in Argobots.

Change-Id: Ief8da1597afded1edc898f847a7bfdf47b59458f
Signed-off-by: Bruno Faccini <bruno.faccini@intel.com>

### CODE REVIEW

**Comment by bfaccini** [comment]
Solo test failure looks strongly related to DAOS-4285 !!...

File: src/vos/vos_common.c [hunk 1]
```diff
@@ -346,6 +378,18 @@ vos_init(void)
...
 
+	/* Some cases (DAOS-4310) have been found where the default ABT's ULTs
+	 * stack-size (16K) is not enough and this can lead to stack overrun
+	 * and corruption, so double it just in case
+	 */
+	if (abt_thread_stacksize() < 32768) {
+		rc = set_abt_thread_stacksize(32768);
+		if (rc != 0) {
+			D_ERROR("failed to set ABT_THREAD_STACKSIZE: %d\n", rc);
+			return rc;
```
Comment (liw) [inline src/vos/vos_common.c]: Forgot to unlock mutex?

File: src/iosrv/init.c [hunk 2]
```diff
@@ -418,6 +418,38 @@ set_abt_max_num_xstreams(int n)
...
+	if (value == NULL)
+		return -DER_NOMEM;
+	D_INFO("Setting %s to %s\n", name, value);
+	rc = setenv(name, value, 1 /* overwrite */);
+	D_FREE(value);
+	if (rc != 0)
+		return daos_errno2der(errno);
+	return 0;
+}
+
```
Comment (liw) [inline src/iosrv/init.c]: [Optional] Could be placed in libgurt to avoid duplication?

Comment (liw) [reply src/iosrv/init.c]: Never mind, sorry. libgurt should not depend on Argobots. There's no good place that could be shared between iosrv and vos.

File: src/iosrv/init.c [hunk 3]
```diff
@@ -418,6 +418,38 @@ set_abt_max_num_xstreams(int n)
...
+	D_ASPRINTF(value, "%d", n);
+	if (value == NULL)
+		return -DER_NOMEM;
+	D_INFO("Setting %s to %s\n", name, value);
+	rc = setenv(name, value, 1 /* overwrite */);
+	D_FREE(value);
+	if (rc != 0)
+		return daos_errno2der(errno);
+	return 0;
+}
```
[the remark this answers is not in the document]
Comment (bfaccini) [reply src/iosrv/init.c]: Well, I have tried but this looks a bit tricky as @liw seems to confirm

File: src/vos/vos_common.c (hunk 1 above)
Comment (bfaccini) [reply src/vos/vos_common.c]: Oops, good catch ! Seems that I have replicated same code from abc_init() a bit too quickly... Will fix in next commit

File: src/iosrv/init.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (bfaccini) [reply src/iosrv/init.c]: Well, I have tried but this looks a bit tricky as @liw seems to confirm

**Pull request merged**